pkgs/by-name/pk/pkl/package.nix

```
@@ -0,0 +1,100 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  gradle,
+  temurin-bin-21,
+  kotlin,
+  nix-update-script,
+  replaceVars,
+  makeWrapper,
+}:
+let
+  jdk = temurin-bin-21;
+  gradleOverlay = gradle.override { java = jdk; };
+  kotlinOverlay = kotlin.override { jre = jdk; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pkl";
+  version = "0.28.2";
+
+  src = fetchFromGitHub {
+    owner = "apple";
+    repo = "pkl";
+    tag = finalAttrs.version;
+    hash = "sha256-5MLSgG1JBxwMKvbp8lZytF7FdBPD0gQSJU+Woh+fCB8=";
+    leaveDotGit = true;
+    postFetch = ''
+      pushd $out
+      git rev-parse --short HEAD > .commit-hash
```

```suggestion
      git rev-parse HEAD > .commit-hash
```
`--short` is not reproducible, the generated hash may get longer if the repository gets more commits.

---

https://github.com/NixOS/nixpkgs/pull/402086#issuecomment-2839209649

---

`--short=8` should always return exactly the first 8 characters, no? Which is what that pkl maintainer in the comment wanted. I guess unless the commit is ambiguous when referenced by only its first 8 characters, which seems to be very rare.